util/argo/argo.go

```diff
@@ -428,3 +467,32 @@ func NormalizeApplicationSpec(spec *argoappv1.ApplicationSpec) *argoappv1.Applic
 	}
 	return spec
 }
+
+func getDestinationServer(ctx context.Context, db db.ArgoDB, clusterName string) (string, error) {
+	clusterList, err := db.ListClusters(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	for _, c := range clusterList.Items {
+		if c.Name == clusterName {
```

Comment:
Currently, cluster name uniqueness is not enforced. Please validate that only one cluster is matching the cluster name.